### PR TITLE
Generate editor keywords

### PR DESCRIPTION
**Pull request by xsebek**
- part of #344

### CODE REVIEW

**Comment by xsebek** [comment]
I updated the VS Code plugin to highlight the new keywords:
https://marketplace.visualstudio.com/items?itemName=xsebek.swarm-language

But it could use a bit more love - update the image link, CHANGELOG,...

File: editors/emacs/swarm-mode.el [hunk 1]
```diff
@@ -25,18 +25,74 @@
...
-             (x-commands
-              '("noop" "wait" "selfdestruct" "move" "turn" "grab" "place" "give"
-                "install" "make" "build" "salvage" "reprogram" "knows"
-                "say" "log" "view" "appear" "create" "getx" "gety"
-                "blocked" "scan" "upload" "ishere" "whoami"
-                "random" "not"
-                "left" "right" "back" "forward" "north" "south" "east" "west" "down"
-                ))
+             (x-builtins '(
+               "if"
```
Comment (xsebek) [inline editors/emacs/swarm-mode.el:32]: @byorgey or @TristanCacqueray, do either of you use EMacs and could check this works, please?

Comment (byorgey) [reply editors/emacs/swarm-mode.el:32]: Works great!

File: editors/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "editors",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {}
+}
```
Comment (xsebek) [inline editors/package-lock.json:6]: Ups, this file should not be here.

File: editors/vscode/syntaxes/swarm.tmLanguage.json [hunk 3]
```diff
@@ -56,7 +56,7 @@
 				},
 				{
 				"name": "keyword.other",
-				"match": "\\b(?i)(noop|wait|selfdestruct|move|turn|grab|place|give|install|make|has|count|drill|build|salvage|reprogram|say|log|view|appear|create|whereami|blocked|scan|upload|ishere|whoami|random|run|if|inl|inr|case|fst|snd|force|return|try|raise|not)\\b"
+				"match": "\\b(?i)(noop|wait|selfdestruct|move|turn|grab|harvest|place|give|install|make|has|count|drill|build|salvage|reprogram|say|log|view|appear|create|whereami|blocked|scan|upload|ishere|self|parent|base|whoami|setname|random|run|if|inl|inr|case|fst|snd|force|return|try|undefined|fail|not|format|teleport|as|robotnamed|robotnumbered|knows)\\b"
```
[the remark this answers is not in the document]
Comment (xsebek) [reply editors/vscode/syntaxes/swarm.tmLanguage.json:59]: Done :heavy_check_mark: 
```
  editors
    VS Code
      operators:                                                        OK
      commands:                                                         OK
      directions:                                                       OK
    EMacs
      builtin:                                                          OK
      commands:                                                         OK
      directions:                                                       OK
```

File: app/Main.hs [hunk 4]
```diff
@@ -41,8 +41,17 @@ cliParser =
...
+    subparser . mconcat $
+      [ command "recipes" (info (pure RecipeGraph) $ progDesc "Output graphviz dotfile of entity dependencies based on recipes")
+      , command "editors" (info (EditorKeywords <$> editor <**> helper) $ progDesc "Output editor keywords")
+      ]
+  editor :: Parser (Maybe EditorType)
+  editor =
+    Data.Foldable.asum
+      [ pure Nothing
+      , Just VSCode <$ switch (long "code" <> help "Generate for the VS Code editor")
+      , Just EMacs <$ switch (long "emacs" <> help "Generate for the EMacs editor")
```
Comment (byorgey) [inline app/Main.hs:53]:
```suggestion
      , Just Emacs <$ switch (long "emacs" <> help "Generate for the Emacs editor")
```

**Review by byorgey — APPROVED** [review]
All looks good, except---and this is being very nitpicky---could we do a search and replace to have `Emacs` instead of `EMacs`?  `Emacs` is never written with a capital `M` and it looks strange to me.

File: app/Main.hs (hunk 4 above)
Comment (xsebek) [reply app/Main.hs]: Hah, sorry about that. :sweat_smile:

**Comment by xsebek** [comment]
Weird, what's up with Mergify? :raised_eyebrow:

**Comment by byorgey** [comment]
This happened to me once before. It took a while but it eventually worked.

File: app/Main.hs (hunk 4 above)
Comment (byorgey) [reply app/Main.hs]: No worries, thanks!

**Pull request merged**